sdk/appcenter-codepush/src/main/java/com/microsoft/appcenter/codepush/utils/CodePushUpdateUtils.java

```
@@ -0,0 +1,316 @@
+            AppCenterLog.error(CodePush.LOG_TAG, e.getMessage(), e);
+            return null;
+        }
+    }
+
+    /**
+     * Returns JWT file path of local update.
+     *
+     * @param updateFolderPath local update directory path.
+     * @return JWT file path of update.
+     */
+    @SuppressWarnings("WeakerAccess")
+    public static String getJWTFilePath(String updateFolderPath) {
+        return FileUtils.appendPathComponent(
+                FileUtils.appendPathComponent(updateFolderPath, CodePushConstants.CODE_PUSH_FOLDER_PREFIX),
+                CodePushConstants.BUNDLE_JWT_FILE_NAME
+        );
+    }
+
+    /**
+     * Returns JWT content of local update.
+     *
+     * @param folderPath local update directory path.
+     * @return JWT content of update.
+     */
+    @SuppressWarnings("WeakerAccess")
+    public static String getJWT(String folderPath) {
+        final String signatureFilePath = getJWTFilePath(folderPath);
+        try {
+            return FileUtils.readFileToString(signatureFilePath);
+        } catch (IOException e) {
+            AppCenterLog.error(CodePush.LOG_TAG, e.getMessage(), e);
+            return null;
+        }
+    }
+
+    /**
+     * Verifies signature of local update.
+     *
+     * @param folderPath      directory of local update.
+     * @param packageHash     remote package hash.
+     * @param stringPublicKey public key value.
+     * @throws CodePushInvalidUpdateException in case of failed signature check.
+     */
+    public static void verifyUpdateSignature(String folderPath, String packageHash, String stringPublicKey) throws CodePushInvalidUpdateException {
+        AppCenterLog.info(CodePush.LOG_TAG, "Verifying signature for folder path: " + folderPath);
+        final PublicKey publicKey = parsePublicKey(stringPublicKey);
+        if (publicKey == null) {
+            throw new CodePushInvalidUpdateException("The update could not be verified because no public key was found.");
```

Do we really need to re-throw these exceptions? I mean we can throw this exception directly from `parsePublicKey` method